pandas/core/dtypes/concat.py

```diff
@@ -68,7 +68,7 @@ def get_dtype_kinds(l):
     return typs
 
 
-def concat_compat(to_concat, axis: int = 0):
+def concat_compat(to_concat, axis: int = 0, ignore_2d_ea: bool = False):
```

Comment:
Will add a doctoring for this later, but the original caller is the only one who knows both the block shape and the extension array shape. So we need to pass that information all the way down to `concat_compat` for whether it's safe to ignore the shape / dim mismatch.